src/IO/ConcatSeekableReadBuffer.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <IO/SeekableReadBuffer.h>
+#include <vector>
+
+
+namespace DB
+{
+
+/// Reads from the concatenation of multiple SeekableReadBuffer's
+class ConcatSeekableReadBuffer : public SeekableReadBufferWithSize
+{
+public:
+    ConcatSeekableReadBuffer() : SeekableReadBufferWithSize(nullptr, 0) { }
+    ConcatSeekableReadBuffer(std::unique_ptr<SeekableReadBuffer> buf1, size_t size1, std::unique_ptr<SeekableReadBuffer> buf2, size_t size2);
```

Comment:
Looks like a very special case... Maybe variadic arguments or a single argument?

Reply:
It is very common to concat two buffers, and we seldom have a necessity to concat 3 or 4 buffers